vyper/parser/parser_utils.py

```diff
@@ -698,37 +704,65 @@ def make_setter(left, right, location, pos, in_function_call=False):
         raise Exception("Invalid type for setters")
 
 
-def decorate_ast(_ast, code, class_names=None):
-    if class_names is None:
-        class_names = {}
+class AnnotatingVisitor(ast.NodeTransformer):
+    _source_code: str
+    _class_types: ClassTypes
 
-    class MyVisitor(ast.NodeTransformer):
-        def visit(self, node):
-            self.generic_visit(node)
+    def __init__(self, source_code: str, class_types: Optional[ClassTypes] = None):
+        self._source_code = source_code
+        if class_types is not None:
+            self._class_types = class_types
+        else:
+            self._class_types = {}
 
-            # Decorate every node of an AST tree with the original source code.
-            # This is necessary to facilitate error pretty-printing.
-            node.source_code = code
+    def generic_visit(self, node):
+        # Decorate every node in the AST with the original source code. This is
+        # necessary to facilitate error pretty-printing.
+        node.source_code = self._source_code
 
-            # Decorate class definition with the type of classes they are.
-            if isinstance(node, ast.ClassDef):
-                node.class_type = class_names.get(node.name)
+        return super().generic_visit(node)
 
-            return node
+    def visit_ClassDef(self, node):
+        self.generic_visit(node)
 
-    MyVisitor().visit(_ast)
+        # Decorate class definitions with their respective class types
+        node.class_type = self._class_types.get(node.name)
 
+        return node
+
+
+class OptimizingVisitor(ast.NodeTransformer):
```

Comment:
I don't think this naming describe what it actually happens OptimizingVisitor is ambiguous, RewriteUnaryOp is clearer name for me.

Reply:
I chose that name because it's a more general term and I thought more such actions could be added to the same transformer in the future. But I'll make the change you're requesting.

Reply:
Yes I think we'll be adding another visitor for constant and type annotations here hopefully soon, but let's evaluate what to name them once we have coded all of those up.